lib/transform-directives-for-browser.js

```diff
@@ -9,36 +36,33 @@ var handlers = {
       var basePolicy = {}
       if (version < 5) {
         basePolicy.allow = ['*']
+
+        if (directives.defaultSrc) {
+          basePolicy.allow = directives.defaultSrc
+          delete directives.defaultSrc
+        }
       } else {
         basePolicy.defaultSrc = ['*']
       }
 
-      return reduce(directives, function (result, value, key) {
-        if (key === 'connectSrc') {
-          result.xhrSrc = value
-        } else if ((key === 'defaultSrc') && (version < 5)) {
-          result.allow = value
-        } else {
-          result[key] = value
-        }
-
-        if (key === 'scriptSrc') {
-          var optionsValues = []
-
-          if (value.indexOf("'unsafe-inline'") !== -1) {
-            optionsValues.push('inline-script')
-          }
-          if (value.indexOf("'unsafe-eval'") !== -1) {
-            optionsValues.push('eval-script')
-          }
+      return createFirefoxPreCSP10Directives(directives, basePolicy, version)
+    } else {
+      return directives
+    }
+  },
 
-          if (optionsValues.length !== 0) {
-            result.options = optionsValues
-          }
-        }
+  'Firefox Mobile': function (browser, directives) {
+    // Handles both Firefox for Android and Firefox OS
+    var basePolicy = {
+      defaultSrc: ['*']
+    }
+    var family = browser.os.family
+    var version = parseFloat(browser.version)
 
-        return result
-      }, basePolicy)
+    if (family === 'Firefox OS' && version < 32) {
```

Comment:
I think I'd prefer this code to look like this:

```js
if ((family === 'Firefox OS' && version < 32) || (family === 'Android' && version < 25)) {
  return createFirefoxPreCSP10Directives(directives, { defaultSrc: ['*'] })
} else {
  // ...
```

Does that seem reasonable?